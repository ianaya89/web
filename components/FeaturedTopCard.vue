<template>
  <div class="relative flex bg-center bg-cover overflow-hidden" :style="{'background-image': 'url('+$get(item,'image',undefined)+')'}">
    <div class="featured-mask" />
    <div class="featured-logo flex items-end ml-12 pb-16 w-32">
      <img :src="$get(item,'logo',undefined)">
    </div>

    <div class="w-3/4 border-r my-2 border-grey py-4 pl-10 pr-6 text-white text-h4 z-10 flex">
      <slot />
    </div>
    <div class="w-1/4 text-white py-6 px-8 z-10">
      <slot name="right" />
    </div>
  </div>
</template>

<script>
export default {
  name: 'FeaturedTopCard',

  props: {
    item: {
      type: Object,
      default: () => {}
    }
  }
}
</script>

<style lang="postcss" scoped>
.featured-mask {
  @apply .absolute .top-0 .left-0 .right-0 .bottom-0;
  background-image: linear-gradient(
    to right,
    rgba(0, 0, 0, 1),
    rgba(0, 0, 0, 0.5)
  );
}

.featured-logo {
  @apply .bg-white .top-0 .bottom-0 .flex .px-4 z-10;
  left: 5rem;
}
</style>
