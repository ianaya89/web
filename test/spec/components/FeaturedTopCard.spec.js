import { shallowMount } from '@vue/test-utils'
import FeaturedTopCard from '@/components/FeaturedTopCard'

describe('FeaturedTopCard', () => {
  it('renders default', () => {
    const wrapper = shallowMount(FeaturedTopCard)
    expect(wrapper.element).toMatchSnapshot()
  })
})
