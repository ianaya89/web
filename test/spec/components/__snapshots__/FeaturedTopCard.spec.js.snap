// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedTopCard renders default 1`] = `
<div
  class="relative flex bg-center bg-cover overflow-hidden"
  style="background-image: url();"
>
  <div
    class="featured-mask"
  />
   
  <div
    class="featured-logo flex items-end ml-12 pb-16 w-32"
  >
    <img
      src=""
    />
  </div>
   
  <div
    class="w-3/4 border-r my-2 border-grey py-4 pl-10 pr-6 text-white text-h4 z-10 flex"
  />
   
  <div
    class="w-1/4 text-white py-6 px-8 z-10"
  />
</div>
`;
